### PR TITLE
NetCore: Replaced nonodes.aspx with MVC controller and view, with configurable view location

### PR DESCRIPTION
**Pull request by AndyButland**
Implements #7707

In this PR, I've removed `nonodes.aspx` and replaced it with a controller accessed via a route.  The controller behaves as the removed file, checking to see if there is any content available, redirecting to root if so, and if not, renders the HTML via a view and view model.

I've added the view location as a configuration parameter as suggested in the issue comments.

And I've done the suggested minor clean-up of the outputted HTML.

### CODE REVIEW

File: src/Umbraco.Core/Constants-Web.cs [hunk 1]
```diff
@@ -34,6 +34,11 @@ public static class Web
             /// The header name that angular uses to pass in the token to validate the cookie
             /// </summary>
             public const string AngularHeadername = "X-UMB-XSRF-TOKEN";
+
+            /// <summary>
+            /// The route for rendering a page when no content is published.
+            /// </summary>
+            public const string NoContentRoute = "/UmbNoContent";
```
Comment (bergmania) [inline src/Umbraco.Core/Constants-Web.cs]:
```suggestion
            /// <summary>
            /// The route name of the page shown when Umbraco has no published content.
            /// </summary>
            public const string NoContentRouteName = "umbraco-no-content";
```

File: src/Umbraco.Web/Runtime/WebInitialComponent.cs [hunk 2]
```diff
@@ -17,7 +17,6 @@
 using Umbraco.Core.Hosting;
 using Umbraco.Core.Strings;
 using Umbraco.Core.IO;
-using Umbraco.Core.Strings;
 using Umbraco.Web.Install;
 using Umbraco.Web.JavaScript;
 using Umbraco.Web.Mvc;
```
Comment (bergmania) [inline src/Umbraco.Web/Runtime/WebInitialComponent.cs]:
```suggestion
using Umbraco.Web.Mvc;
using Constants = Umbraco.Core.Constants;
```

File: src/Umbraco.Web/Runtime/WebInitialComponent.cs [hunk 3]
```diff
@@ -191,6 +193,14 @@ private void ConfigureClientDependency()
             RoutePluginControllers(globalSettings, surfaceControllerTypes, apiControllerTypes, ioHelper);
         }
 
+        private static void RouteNoContentController(string umbracoPath)
+        {
+            RouteTable.Routes.MapRoute(
+                "umbraco-no-content",
+                umbracoPath + Core.Constants.Web.NoContentRoute,
+                new { controller = "RenderNoContent", action = "Index" });
```
Comment (bergmania) [inline src/Umbraco.Web/Runtime/WebInitialComponent.cs]:
```suggestion
            RouteTable.Routes.MapRoute(
                Constants.Web.NoContentRouteName,
                umbracoPath + "/UmbNoContent",
                new { controller = "RenderNoContent", action = "Index" });
```

File: src/Umbraco.Web/UmbracoInjectedModule.cs [hunk 4]
```diff
@@ -244,8 +241,8 @@ private bool EnsureHasContent(IUmbracoContext context, HttpContextBase httpConte
 
             _logger.Warn<UmbracoModule>("Umbraco has no content");
 
-            const string noContentUrl = "~/config/splashes/noNodes.aspx";
-            httpContext.RewritePath(_uriUtility.ToAbsolute(noContentUrl));
+            var rewriteTo = _uriUtility.ToAbsolute(_globalSettings.UmbracoPath + Constants.Web.NoContentRoute);
+            httpContext.RewritePath(_uriUtility.ToAbsolute(rewriteTo));
```
Comment (bergmania) [inline src/Umbraco.Web/UmbracoInjectedModule.cs]:
```suggestion
            if (RouteTable.Routes[Constants.Web.NoContentRouteName] is Route route)
            {
                httpContext.RewritePath(route.Url);
            }
```

**Comment by AndyButland** [comment]
Looks good to me @bergmania - I've checked the changes locally and they work fine, so have committed to the PR.

File: src/Umbraco.Web/UmbracoInjectedModule.cs (hunk 4 above)
Comment (AndyButland) [reply src/Umbraco.Web/UmbracoInjectedModule.cs]: Have updated to this.

File: src/Umbraco.Web/Runtime/WebInitialComponent.cs (hunk 3 above)
Comment (AndyButland) [reply src/Umbraco.Web/Runtime/WebInitialComponent.cs]: Have updated to this.

File: src/Umbraco.Web/Runtime/WebInitialComponent.cs (hunk 2 above)
Comment (AndyButland) [reply src/Umbraco.Web/Runtime/WebInitialComponent.cs]: Have updated to this.

File: src/Umbraco.Core/Constants-Web.cs (hunk 1 above)
Comment (AndyButland) [reply src/Umbraco.Core/Constants-Web.cs]: Have updated to this.

**Comment by Shazwazza** [comment]
Looks great! 🎉 @AndyButland any chance you can fix up the merge issues and then i'll merge in.

**Comment by AndyButland** [comment]
Sure, that's updated now.

**Pull request merged**